Make default codegen behavior skip Lower function

wconstab opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #75274
* __->__ #75267
* #75264

- clean up arguments relating to ts backend generation
- make entire lowering function rather than just body be a part of
  backend-IR class

Differential Revision: [D35411212](https://our.internmc.facebook.com/intern/diff/D35411212)

wconstab committed:
Make default codegen behavior skip Lower function

- clean up arguments relating to ts backend generation
- make entire lowering function rather than just body be a part of
  backend-IR class

[ghstack-poisoned]

wconstab commented:
@antoniojkim Could you adapt to this change on your end? We need to somehow make it possible for XLA team to avoid generating a Lower function for now.

antoniojkim commented:
> @antoniojkim Could you adapt to this change on your end? We need to somehow make it possible for XLA team to avoid generating a Lower function for now.

yep, I can make this work. Thanks for letting me of it

bdhirsh reviewed (approved):
lgtm

JackCaoG reviewed (approved):
Thanks!

wconstab commented:
@wconstab has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35411212).

wconstab committed:
Update on "Make default codegen behavior skip Lower function"


- clean up arguments relating to ts backend generation
- make entire lowering function rather than just body be a part of
  backend-IR class

Differential Revision: [D35411212](https://our.internmc.facebook.com/intern/diff/D35411212)

[ghstack-poisoned]

wconstab commented:
@wconstab has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35411212).

wconstab committed:
Update on "Make default codegen behavior skip Lower function"


- clean up arguments relating to ts backend generation
- make entire lowering function rather than just body be a part of
  backend-IR class

Differential Revision: [D35411212](https://our.internmc.facebook.com/intern/diff/D35411212)

[ghstack-poisoned]

wconstab commented:
@wconstab has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35411212).
